backend/app.js

```diff
@@ -43,4 +43,32 @@ app.post('/profile', (req, res) => {
     res.status(200).send("Profile endpoint under construction.");
 });
 
+app.get('/reset-password/:username', async (req, res) => {
+    try {
+        const { username } = req.params;
+        const { newPassword } = req.body;
+
+        if (!username) {
+            throw new Error("Please log in to reset password");
+        };
+
+        const userInfo = await prisma.user.findUnique({
+            where: { username: username },
```

Comment:
I'm not sure why this isn't showing up as an error in your code, but the `username` field in your prisma schema is not unique, therefore `prisma.user.findUnique` would not work here. 

If you apply my suggested code changes above, then your prisma.user.findUnique() should be querying the userId instead of the username because the userId is unique.

```suggestion
        const userInfo = await prisma.user.findUnique({
            where: { userId  },
```